Sources/App/Core/CurrentReferenceCache.swift

```diff
@@ -0,0 +1,20 @@
+import Cache
+
+
+typealias CurrentReferenceCache = ExpiringCache<String, String>
+
+
+extension CurrentReferenceCache {
+    static let live = CurrentReferenceCache(duration: .minutes(5))
```

Comment:
I don't love this but I don't see another way for us to deal with this than introduce a short-lived in-memory cache of owner/repo → reference mappings.

Reply:
I haven't checked into how these additional queries work yet, but are they only on the first JavaScript page load?

11 extra queries on one documentation page that then facilitates any further browsing without requests might be preferable to introducing this cache? If it's on every JavaScript navigation event, I agree this will be necessary, but if it's just the first load, we should weigh this trade off up.

Reply:
It's 19 db requests on initial documentation page load. I believe subsequent loads do not hit the db but even so we really don't want all those unnecessary requests hitting the db.

Reply:
But doesn't this way mean that there will be a 5 minute period where documentation will be broken after analysis runs and finds a new release? That's a big tradeoff.

Reply:
Ah, I remember now, we already have something that serves up old documentation until the builds finish, so this would be no worse than that.

Reply:
I believe the build delay would take care of that, plus at worst we'd be mixing css/js that's probably the same between revisions. We could also purge the cache for the specific repo when we discover the new version.

Reply:
(re old docs vs just resource - we wouldn't _have_ to use the cache for the docs but I thought we might as well)